src/architecture/utilities/tests/test_orbitalMotion.cpp

```diff
@@ -0,0 +1,533 @@
+}
+
+TEST(OrbitalMotion, solarRadiationPressure) {
+    double check[3];
+    double ans[3];
+    double A = 2.0;
+    double m = 50.0;
+    double r[3] = {1.0, 0.3, -0.2};
+
+    v3Set(-1.9825487816e-10, -5.94764634479e-11, 3.96509756319e-11, check);
+    solarRad(A, m, r, ans);
+    EXPECT_TRUE(v3IsEqual(ans, check, 11));
+}
+
+TEST(OrbitalMotion, elem2rv1DEccentric)
+{
+    double r[3];
+    double v[3];
+    double r2[3];
+    double v3_2[3];
+    classicElements elements;
+    elements.a = 7500.0;
+    elements.e = 1.0;
+    elements.i = 40.0 * D2R;
+    elements.Omega = 133.0 * D2R;
+    elements.omega = 113.0 * D2R;
+    elements.f = 23.0 * D2R;
+    elements.alpha = 1.0 / elements.a;
+    elements.rPeriap = elements.a*(1.-elements.e);
+    elements.rApoap = elements.a*(1.+elements.e);
+    elem2rv(MU_EARTH, &elements, r, v);
+    v3Set(-148.596902253492, -457.100381534593, 352.773096481799, r2);
+    v3Set(-8.93065944520745, -27.4716886950712, 21.2016289595043, v3_2);
+    EXPECT_TRUE(v3IsEqualRel(r, r2, orbitalElementsAccuracy) || !v3IsEqualRel(v, v3_2, orbitalElementsAccuracy));
+}
+
+TEST(OrbitalMotion, elem2rv1DHyperbolic)
+{
+    double r[3];
+    double v[3];
+    double r2[3];
+    double v3_2[3];
+    classicElements elements;
+    elements.a = -7500.0;
+    elements.e = 1.0;
+    elements.i = 40.0 * D2R;
+    elements.Omega = 133.0 * D2R;
+    elements.omega = 113.0 * D2R;
+    elements.f = 23.0 * D2R;
+    elements.alpha = 1.0 / elements.a;
+    elements.rPeriap = elements.a*(1.-elements.e);
+    elements.rApoap = elements.a*(1.+elements.e);
+    elem2rv(MU_EARTH, &elements, r, v);
+    v3Set(-152.641873349816, -469.543156608544, 362.375968124408, r2);
+    v3Set(-9.17378720883851, -28.2195763820421, 21.7788208976681, v3_2);
+    EXPECT_TRUE(v3IsEqualRel(r, r2, orbitalElementsAccuracy) && v3IsEqualRel(v, v3_2, orbitalElementsAccuracy));
+}
+
+class TwoDimensionHyperbolic : public ::testing::Test {
+protected:
+    double r[3];
+    double v[3];
+    double r2[3];
+    double v3_2[3];
+    classicElements elements;
+
+    void SetUp() override {
+        elements.a = -7500.0;
+        elements.e = 1.4;
+        elements.i = 40.0 * D2R;
+        elements.Omega = 133.0 * D2R;
+        elements.omega = 113.0 * D2R;
+        elements.f = 23.0 * D2R;
+        elements.alpha = 1.0 / elements.a;
+        elements.rPeriap = elements.a*(1.-elements.e);
+        elements.rApoap = elements.a*(1.+elements.e);
+
+        v3Set(319.013136281857, -2796.71958333493, 1404.6919948109, r2);
+        v3Set(15.3433051336115, -5.87012423567412, -6.05659420479213, v3_2);
+    }
+};
+
+TEST_F(TwoDimensionHyperbolic, elem2rv)
+{
+    elem2rv(MU_EARTH, &elements, r, v);
+    EXPECT_TRUE(v3IsEqualRel(r, r2, orbitalElementsAccuracy) && v3IsEqualRel(v, v3_2, orbitalElementsAccuracy));
+}
+
+TEST_F(TwoDimensionHyperbolic, rv2elem) {
+    rv2elem(MU_EARTH, r2, v3_2, &elements);
+    EXPECT_PRED3(isEqualRel, elements.a, -7500.0, orbitalElementsAccuracy);
```

Comment:
Perhaps instead of:
```
EXPECT_PRED3(isEqualRel, x, y, relAccuracy);
```
we define a new macro in "unitTestComparators.h" to support this syntax:
```
EXPECT_NEAR_REL(x, y, relAccuracy)
```
which parallels `EXPECT_NEAR(x, y, absAccuracy)`. 

This can also apply to vector comparison. Instead of `EXPECT_PRED` with `v3IsEqualRel`, we use:
```
EXPECT_VECTOR3_NEAR(aVec, bVec, absTol)
```
```
EXPECT_VECTOR3_NEAR_REL(aVec, bVec, relTol)
```

Reply:
Agreed. I was thinking that I'd port the tests faithfully and do these kinds of refactors in a separate PR. As opposed to asking reviewers to review this addition and then also carefully validate the tests assert changes. But maybe this is unnecessary caution. I'm happy to add it here if you think it's not worth splitting up.

Reply:
No, you are right, best to err in the side of more atomic PRs than in the side of monolithic harder-to-review PRs. We'll open an new issue with custom features/utilities we would like to see added for Google Tests and do a well thought out implementation.

Reply:
I added #328 and #329 to follow up on these thoughts. Feel free to add/edit with any additional thoughts you have 😄